api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/ApiGatewayControllerTest.java

```diff
@@ -206,38 +243,74 @@ void deleteUser() {
         assertEquals(null, authServiceClient.getUser(user.getId()));
     }
 
-//    @Test
-//    void getPutRequestOk()  {
-//
-//        OwnerDetails od = new OwnerDetails();
-//        od.setId(1);
-//        od.setFirstName("John");
-//        od.setLastName("Doe");
-//        od.setAddress("1 Star Street");
-//        od.setCity("Boston");
-//        od.setTelephone("5553451125");
-//
-//        Mockito.when(customersServiceClient.updateOwner(od,1))
-//                .thenReturn(Mono.just(od));
-//
-//        client.put()
-//                .uri("api/gateway/owners/{ownerId}", 1)
-//                .contentType(MediaType.APPLICATION_JSON)
-//                .accept(MediaType.APPLICATION_JSON)
-//                .exchange()
-//                .expectStatus().isOk()
-//                .expectBody();
-//
-//
-//        assertEquals(od.getId(), 1);
-//        assertEquals(od.getFirstName(), "John");
-//        assertEquals(od.getLastName(),"Doe");
-//        assertEquals(od.getAddress(), "1 Star Street");
-//        assertEquals(od.getCity(), "Boston");
-//        assertEquals(od.getTelephone(),"5553451125");
-//
-//
-//    }
+
+
+    //private static final int BILL_ID = 1;
+
+    @Test
+    public void getBillById(){
+
+        //int expectedLength = 1;
+
+        BillDetails entity = new BillDetails();
+
+        entity.setBillId(1);
+
+        entity.setAmount(599);
+
+        entity.setCustomerId(2);
+
+        entity.setVisitType("Consultation");
+
+        when(billServiceClient.getBilling(1))
+                .thenReturn(Mono.just(entity));
+
+        client.get()
+                //check the URI
+                .uri("/api/gateway/bills/1")
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody()
+                .jsonPath("$.billId").isEqualTo(1)
+                .jsonPath("$.customerId").isEqualTo(entity.getCustomerId())
+                .jsonPath("$.visitType").isEqualTo(entity.getVisitType())
+                .jsonPath("$.amount").isEqualTo(entity.getAmount());
+
+
+
+
+        assertEquals(entity.getBillId(), 1);
```

Comment:
This is questionable, you are just getting what you have set and comparing it but ok.